packages/core/src/components/modal/modal.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { newSpecPage, SpecPage } from '@stencil/core/testing'
+
+import { AtomModal } from './modal'
+
+describe('atom-modal', () => {
+  let page: SpecPage
+
+  beforeEach(async () => {
+    page = await newSpecPage({
+      components: [AtomModal],
+      html: `
+        <atom-modal
+          trigger="button"
+          primary-text="Primary"
+          secondary-text="Secondary" 
+        >
+          Modal content
+        </atom-modal>`,
+    })
+  })
+
+  it('should render header slot when headerTitle is not passed', async () => {
+    expect(page.root).toEqualHtml(`
+      <atom-modal primary-text="Primary" secondary-text="Secondary" trigger="button">
+        <ion-modal aria-describedby="atom-modal__content" aria-labelledby="atom-modal__header-title" class="atom-modal" trigger="button">
+          <header class="atom-modal__header">
+            <div id="atom-modal__header-title"></div>
+            <atom-button aria-label="close" class="atom-modal__close" fill="clear" shape="circle">
+              <atom-icon aria-hidden="true" class="atom-modal__close-icon" icon="close"></atom-icon>
+            </atom-button>
+          </header>
+          <div class="atom-modal__content" id="atom-modal__content">
+            Modal content
+          </div>
+          <footer class="atom-modal__footer">
+            <atom-button class="atom-modal__btn-action atom-modal__btn-action--secondary" color="secondary" fill="outline">
+              Secondary
+            </atom-button>
+            <atom-button class="atom-modal__btn-action atom-modal__btn-action--primary" color="primary">
+              Primary
+            </atom-button>
+          </footer>
+        </ion-modal>
+      </atom-modal>
+    `)
+  })
+
+  it('should render header slot when headerTitle is not passed', async () => {
```

Comment:
this title is repeated, man

Reply:
my bad, done 8f6b550ff857488517b5b29d54c47fdb80fa5d62